katportalclient/client.py

```diff
@@ -131,6 +172,78 @@ def __init__(self, url, on_update_callback, io_loop=None, logger=None):
         self._ws_jsonrpc_cache = []
         self._heart_beat_timer = PeriodicCallback(
             self._send_heart_beat, WS_HEART_BEAT_INTERVAL)
+        self._current_user_id = None
+
+    @tornado.gen.coroutine
+    def logout(self):
+        """ Logs user out of katportal. Katportal then deletes the cached
+        session_id for this client. In order to call HTTP requests that
+        requires authentication, the user will need to login in again.
+        """
+        try:
+            url = self.sitemap['authorization'] + '/user/logout'
+            response = yield self.authorized_fetch(
+                url=url, auth_token=self._session_id, method='POST', body='{}')
```

Comment:
Good point, if you don't have a local `_session_id`, then there's no point to calling the katportal endpoint. Added a check in ad99d45